File: db/java/app/src/main/java/com/example/sales/domain/model/PriceByCustomer.java
package com.example.sales.domain.model;

import java.time.LocalDateTime;

/**
 * 顧客別販売単価のEntityクラス
 */
public class PriceByCustomer {
    private String productCode;
    private String customerCode;
    private Integer sellingPrice;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;

    // デフォルトコンストラクタ
    public PriceByCustomer() {}

    // Getter/Setter
    public String getProductCode() { return productCode; }
    public void setProductCode(String productCode) { this.productCode = productCode; }

    public String getCustomerCode() { return customerCode; }
    public void setCustomerCode(String customerCode) { this.customerCode = customerCode; }

    public Integer getSellingPrice() { return sellingPrice; }
    public void setSellingPrice(Integer sellingPrice) { this.sellingPrice = sellingPrice; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public String getCreatedBy() { return createdBy; }
    public void setCreatedBy(String createdBy) { this.createdBy = createdBy; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }

    public String getUpdatedBy() { return updatedBy; }
    public void setUpdatedBy(String updatedBy) { this.updatedBy = updatedBy; }
}

File: db/go/sales_management/internal/model/price_by_customer.go
package model

import "time"

// PriceByCustomer は顧客別販売単価のドメインモデルです
type PriceByCustomer struct {
	ProductCode  string    `db:"商品コード"`
	CustomerCode string    `db:"取引先コード"`
	Price        int       `db:"販売単価"`
	CreatedAt    time.Time `db:"作成日時"`
	CreatedBy    string    `db:"作成者名"`
	UpdatedAt    time.Time `db:"更新日時"`
	UpdatedBy    string    `db:"更新者名"`
}
